x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -934,6 +1052,39 @@ private static Map<String, Object> maybeRewriteMetadataForApiKeyRoleDescriptors(
         return metadata;
     }
 
+    // pkg-private for testing
+    static Map<String, Object> maybeRewriteMetadataForRemoteAccessAuthentication(
```

Comment:
This is based on part of our discussion around BWC in your review comment. 

The rewrite logic here is provisional: for 8.7, we won't ever end up rewriting since remote access authentication cannot be rewritten towards anything older than 8.7. However, it seems prudent to have the logic in place now -- it's a loaded foot gun otherwise as we move into versions past 8.7, and possibly change the authentication model. I'm okay also holding off with introducing this in 8.8, as it makes testing less awkward. Let me know what you prefer.